### PR TITLE
30) added new controllers, and connected with it staff

### PR DESCRIPTION
**Pull request by AntonPanchishko**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userId) {
+        return orderMapper
+                .toDtoFromObject(orderService
+                        .completeOrder(shoppingCartService
+                                .getByUser(userService
+                                        .getById(userId))));
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: Let's not do this ladder))

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public List<OrderResponseDto> getAllOrder(@RequestParam Long userId) {
+        return orderService
+                .getOrdersHistory(userService.getById(userId))
+                .stream()
+                .map(orderMapper::toDtoFromObject)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/complete")
+    public OrderResponseDto completeOrder(@RequestParam Long userId) {
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/controller/OrderController.java:46]: If I'm not mistaken, this must be POST method

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                           UserService userService,
+                           ShoppingCartService shoppingCartService) {
+        this.orderService = orderService;
+        this.orderMapper = orderMapper;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getAllOrder(@RequestParam Long userId) {
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/controller/OrderController.java]:
```suggestion
    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
```

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                                  MovieSessionService movieSessionService,
+                                  ShoppingCartMapper shoppingCartMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartMapper = shoppingCartMapper;
+    }
+
+    @PostMapping
+    public void add(@RequestParam Long userId, Long movieSessionId) {
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]:
```suggestion
    public void add(@RequestParam Long userId, @RequestParam Long movieSessionId) {
```
It won't work in other way

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -56,4 +56,13 @@ public User add(User user) {
                     + email, e);
         }
     }
+
+    @Override
+    public Optional<User> getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(User.class, id));
+        } catch (Exception e) {
+            throw new DataBindingException("Can't find entity by id " + id, e);
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]:
```suggestion
            throw new DataBindingException("Can't find user by id " + id, e);
```

File: src/main/java/com/dev/cinema/model/dto/response/OrderResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
+package com.dev.cinema.model.dto.response;
+
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/model/dto/response/OrderResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/com/dev/cinema/model/dto/response/ShoppingCartResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package com.dev.cinema.model.dto.response;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/model/dto/response/ShoppingCartResponseDto.java]: same

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/AuthenticationController.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService service;
+
+    public AuthenticationController(AuthenticationService service) {
+        this.service = service;
+    }
+
+    @PostMapping("/register")
+    public void registration(@RequestBody UserRequestDto requestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/AuthenticationController.java]:
```suggestion
    public void register(@RequestBody UserRequestDto requestDto) {
```
method names should be verbs

File: src/main/java/com/dev/cinema/controller/AuthenticationController.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import com.dev.cinema.security.AuthenticationService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService service;
+
+    public AuthenticationController(AuthenticationService service) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/AuthenticationController.java:15]: let's use annotation above the constructor everywhere

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+                                  MovieSessionService movieSessionService,
+                                  ShoppingCartMapper shoppingCartMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartMapper = shoppingCartMapper;
+    }
+
+    @PostMapping
+    public void add(@RequestParam Long userId, @RequestParam Long movieSessionId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]:
```suggestion
    public void addSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
```

File: src/main/java/com/dev/cinema/service/mapper/OrderMapper.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
+package com.dev.cinema.service.mapper;
+
+import com.dev.cinema.model.Order;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.dto.response.OrderResponseDto;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/OrderMapper.java]:
```suggestion
@Component
```
same for all mappers

File: src/main/java/com/dev/cinema/service/mapper/UserMapper.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
+package com.dev.cinema.service.mapper;
+
+import com.dev.cinema.model.User;
+import com.dev.cinema.model.dto.request.UserRequestDto;
+import com.dev.cinema.model.dto.response.UserResponseDto;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserMapper {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/UserMapper.java]: redundant empty line

File: src/main/java/com/dev/cinema/service/mapper/UserMapper.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    public User fromDtoToObject(UserRequestDto userRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/UserMapper.java]: do we need this method?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**